### PR TITLE
Upgrade bug - Yum cache update not running

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Simple fix to force the yum cache to be cleared, sometimes handler not run

Replaces handlers with conditional task run bc of https://github.com/ansible/ansible/issues/41313

Fixes # (https://github.com/confluentinc/cp-ansible/issues/495)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

@dainesj Tested the custom repo upgrade path

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.common/tasks/redhat.yml [hunk 1]
```diff
@@ -20,17 +22,24 @@
     gpgcheck: "{{confluent_common_repository_redhat_main_gpgcheck}}"
     gpgkey: "{{confluent_common_repository_redhat_main_gpgkey}}"
     enabled: "{{confluent_common_repository_redhat_main_enabled}}"
-  notify: yum-clean-all
-  when: not custom_yum_repofile|bool
+  register: confluent_repo_result
+  when:
+    - repository_configuration == 'confluent'
+    - installation_method == "package"
```
Comment (domenicbove) [inline roles/confluent.common/tasks/redhat.yml:28]: installation method in this version?

**Review by dainesj — APPROVED** [review]
Validated RHEL changes in AWS :
Linux ip-172-31-88-54.ec2.internal 4.18.0-240.1.1.el8_3.x86_64 #1 SMP Fri Oct 16 13:36:46 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux

**Review by JumaX — APPROVED** [review]
LGTM.

**Comment by JumaX** [comment]
@domenicbove should be merged into 5.5.x.  All -post branches are frozen for all intents and purposes.

**Comment by JumaX** [comment]
@domenicbove ignore my last comment.  It does make sense for this to go into the -post branches, as it is a critical bug fix, and we don't have another way to hotfix this.  It would also help our customers the most, which at the end of the day, is the whole point. :)

**Pull request merged**